activerecord/lib/active_record/signed_id.rb

```diff
@@ -60,6 +60,8 @@ def find_signed(signed_id, purpose: nil)
       #   User.first.destroy
       #   User.find_signed! signed_id # => ActiveRecord::RecordNotFound
       def find_signed!(signed_id, purpose: nil)
+        raise UnknownPrimaryKey.new(self) if primary_key.nil?
```

Comment:
Does `find_signed!` not work as expected without this line? https://github.com/rails/rails/pull/39542#discussion_r435644023

Reply:
A test case that works for `find_signed` doesn't work for `find_signed!`:
```ruby
  test "find signed record with a bang raises UnknownPrimaryKey when a model has no primary key" do
    error = assert_raises(ActiveRecord::UnknownPrimaryKey) do
      Matey.find_signed!("this will not be even verified")
    end
    assert_equal "Unknown primary key for table mateys in model Matey.", error.message
  end
```
So what I wanted to achieve is symmetry in behavior between those tests, but now I feel that this change makes no sense, and doesn't solve any issue as per https://github.com/rails/rails/pull/39542#discussion_r435644023.
I gonna remove this change and leave this PR with new tests for `find_signed!` to make sure that `find_signed!` works with custom primary key and others.